app/main/controllers/media/RTMediaComment.php

```diff
@@ -99,6 +100,83 @@ function remove( $id ) {
 				return $comment_deleted;
 			}
 		}
-				return false;
+		return false;
+	}
+
+
+	/**
+	 * Helper function to check whether the shortcode should be rendered or not
+	 *
+	 * @return bool
+	 */
+	static function display_allowed() {
+		global $rtmedia_query;
+		$flag = ( ! (  is_home() || is_post_type_archive() || is_author()))
+		&& is_user_logged_in()
+		&& (is_rtmedia_upload_music_enabled() || is_rtmedia_upload_photo_enabled() || is_rtmedia_upload_video_enabled())
+		 //added condition to disable upload when media is disabled in profile/group but user visits media tab
+		&& ( ( isset( $rtmedia_query->is_upload_shortcode ) && true === $rtmedia_query->is_upload_shortcode )
+				|| ( is_rtmedia_bp_profile() && is_rtmedia_profile_media_enable() )
+				||  (is_rtmedia_bp_group() && is_rtmedia_group_media_enable()) );
+		$flag = apply_filters( 'before_rtmedia_uploader_display', $flag );
```

Comment:
Change the filter name. This `before_rtmedia_uploader_display` filter is already being used by `rtmedia_uploader` shortcode